src/map.rs

```diff
@@ -457,6 +457,37 @@ where
         }
     }
 
+    /// Returns a reference to the value with the given `hash` and which
+    /// satisfies the equality function passed.
+    ///
+    /// This function is similar to [`Self::get()`], but provides a shortcut
+    /// for lookups where instantiation of keys is expensive. For instance, if
+    /// creating the key requires allocation, this can be avoided by using
+    /// this method.
+    ///
+    /// As a caller, you are responsible for using the correct hasher, and
+    /// making sure the equality function only returns `true` for the key you
+    /// are looking for.
+    #[inline]
+    pub fn get_by_hash<'g, Q>(
+        &self,
+        hash: u64,
+        eq: impl Fn(&Q) -> bool,
```

Comment:
I think this just be `Fn(&K) -> bool` and we can remove `Q` entirely? We could also remove the `K: Hash` restriction on this method that comes from the outer impl block, but I suppose that wouldn't be very useful without a similar insertion method (or `HashTable` interface).